Convert all remaining tests in the Loader subtree to the merged model

trylek opened the pull request:
/cc @dotnet/runtime-infrastructure

trylek committed:
Manual modifications to tests under Loader/classloader/generics

trylek committed:
Mechanically merge all tests under Loader/classloader/generics

trylek committed:
Fix two overlooked renames of the Methods/Lib.ilproj library project

trylek committed:
Manual modifications to tests under Loader/classloader/regressions

trylek committed:
Mechnical merging of all tests under Loader/classloader/regressions

trylek committed:
Manual modifications to remaining tests under Loader/classloader

trylek committed:
Mechanically merge all remaining tests under Loader/classloader

trylek committed:
Shorten several excessively long names causing trouble even with long paths enabled

trylek committed:
Revert changes to TSAmbiguities test and mark them with explicit Main

Visibility modifications to these tests tamper with their purpose
to test various corner typesystem cases. As we're using them out of
process anyway we can easily keep the original visibilities.

Thanks

Tomas

trylek committed:
Manual modifications to remaining tests under Loader

trylek committed:
Mechanically merge all remaining tests under Loader

trylek committed:
Mark the two Unloaded projects as DisableProjectBuild to suppress duplicity complaint

trylek committed:
Fix annotation for the Unloaded library projects

markples commented on src/tests/Loader/CollectibleAssemblies/ByRefLocals/Unloaded.csproj:
```diff
@@ -2,7 +2,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Library</OutputType>
-    <DisableProjectBuild>true</DisableProjectBuild>
+    <RequiresProcessIsolation>true</RequiresProcessIsolation>
```
Library and RPI seem contradictory (though I think this is the case where Library is the default but one of our files overwrites it when RPI is set)

trylek replied:
I agree it's somewhat dirty but it was the easiest way to suppress the "duplicate project name" error - I can rename the two projects if you find that preferable but it will require changes to the test themselves as their source code hardcodes the assembly name. I'll be happy for any suggestions how to address this in a cleaner way.

markples replied:
Oh.. can you confirm that this is what is happening - 

ByRefLocals and Statics both have (unrelated) libraries called Unloaded.  They work because they are marked RPI and get different directories and then msbuild copies their references into those directories.  (Otherwise, the merged test group would have two Unloaded libraries.)  However, the merged test group would still get two Unloaded libraries (the duplicates error), so this stops them from being included in that directory.  These are likely getting simple wrappers generated with zero actual tests in them (weird but harmless).

I'm definitely not sure about this, but perhaps they could be removed from the overall MergedWrapperProjectReference.  This may be a lurking problem if RPI is removed from the actual tests here (though maybe that's not possible due to the tests themselves) or if the output directory structure changes, so renaming one might be useful anyway.

trylek replied:
OK, thanks for your feedback, I agree with you that it's probably going to be cleanest to deduplicate the name of the Unloaded library, I'll do that as the next commit.

trylek replied:
Fixed in 14th commit.

markples commented on src/tests/Loader/CollectibleAssemblies/Statics/Unloaded.csproj:
```diff
@@ -2,7 +2,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Library</OutputType>
-    <DisableProjectBuild>true</DisableProjectBuild>
+    <RequiresProcessIsolation>true</RequiresProcessIsolation>
```
same

trylek replied:
Fixed in 14th commit.

trylek committed:
Deduplicate the Unloaded support assemblies per Mark's PR feedback

trylek commented:
/azp run runtime-coreclr outerloop

trylek committed:
Fix the ResolvedFromDifferentContext collectible assembly test

This test uses non-local references to dependent projects
TestClass and TestInterface. I have modified the project so that
it copies the two managed assemblies into subdirectories of the
test output folder so that Helix publishing properly picks them
up along with the test.

Thanks

Tomas

trylek commented:
/azp run runtime-coreclr outerloop

trylek committed:
Remove superfluous imports of Directory.Merged.props from Loader subfolders

trylek commented:
/azp run runtime-coreclr outerloop

trylek committed:
Fix BinderTracingTest filtering on Mono

trylek commented:
Mono errors in the recent runs were caused by a subtle bug in the method ```GetFilteredExcludeList``` at the end of ```issues.targets``` - in the presence of files with multiple dots in the name ("multiple extensions") is was dropping everything after the first dot. I have pushed the fix to issues.targets as the last commit on this PR, outerloop is green, I plan to merge this in if the Mono legs end up green after this change (not too hopeful, something seems to be crashing the LLVMAOT compiler, I'll look into that once the job finishes).

trylek committed:
Fix genmeth.ilproj to indicate this is a build-only project

trylek committed:
Fix LLVAM AOT crash when compiling the genmeth.dll assembly

trylek commented:
/azp run runtime-coreclr outerloop

trylek commented:
/azp run runtime-extra-platforms

trylek commented:
Outerloop is green and runtime-extra-platforms doesn't seem to fare much worse than in a recent Steve Toub's PR so I'm about to merge this in.
